### PR TITLE
feat(demo): Add carousel demo code

### PR DESCRIPTION
**Pull request by JaeYeopHan**
* Add carousel demo code
  * What should I add to the contents of the pull request?

### CODE REVIEW

File: demo/assets/js/carousel.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    const detailViewAxes = new eg.Axes({
+        axis: {
+            X: {
+                range: [-50, 0]
+            },
+            Y: {
+                range: [-100, 0]
+            }
+        },
+        bounce: [0, 0],
```
Comment (sculove) [inline demo/assets/js/carousel.js]: The bounce option must be applied to the axis like this
```js
X: {
   range: [ -50, 0],
   bounce: [ 0, 0]
}
```
the default value of bounce is `[0,0]`. so you can ignore it.

File: demo/assets/js/carousel.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            detailViewAxes.option().axis.Y.range[1] = 0;
+        } else {
+            detailViewAxes.option().axis.X.range[0] -= rangeRatio;
+            detailViewAxes.option().axis.X.range[1] += rangeRatio;
+            detailViewAxes.option().axis.Y.range[0] -= rangeRatio;
+            detailViewAxes.option().axis.Y.range[1] += rangeRatio;
+        }
+    }
+
+    const carouselAxes = new eg.Axes({
```
Comment (sculove) [inline demo/assets/js/carousel.js]: You use just one Axes instance
```js
new eg.Axes({
   // ...
   axis: {
      detailX : { range : [-50, 0] }
      detailY : { range : [-100, 0] }
      carousel : { range: [BASE - (PANEL_WIDTH * (LAST_PANEL_IDX - 1)), BASE] }
   }
   // ...
}).mapInput(["detailX", "detailY"], hammerInput1)
  .mapInput(["carousel"], hammerInput2);
```

File: demo/assets/js/carousel.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+	Array.from(document.querySelectorAll("[data-boxId]")).forEach((elm) => {
+        const boxId = elm.dataset.boxid;
+        imgSrcCache[boxId] = elm.children[0].src;
+        boxElmCache[boxId] = elm;
+    });
+
+    const detailViewAxes = new eg.Axes({
+        axis: {
+            X: {
+                range: [-50, 0]
```
Comment (sculove) [inline demo/assets/js/carousel.js]: If you set scale to negative, you can use the range as a positive value.
```js
range: [0, 50]
scale: [-0.4, -0.4]
```

File: demo/assets/js/carousel.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        })
+    ).setTo({
+        X: -25,
+		Y: 50
+	});
+
+    const adjustRange = function(scaleRatio = 1, direction = 1) {
+        let rangeRatio = Math.floor((scaleRatio - 1) * CONTENT_VIEW_PORT / 4 * direction);
+        if (scaleRatio === 1) {
+            detailViewAxes.option().axis.X.range[0] = -50;
```
Comment (sculove) [inline demo/assets/js/carousel.js]: you can access property like this.
```js
const axisOptions = detailViewAxes.options.axis;

axisOptions.X
```

File: demo/assets/js/carousel.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+	const carouselArea = document.querySelector("#carousel-area");
+	const detailViewImg = document.querySelector("#detail-view-content > img");
+    const btnMoreInfo = document.querySelector("#btn-more-info");
+    const detailViewArea = document.querySelector("#detail-view-carousel");
+    
+    let scaleRatio = 1;
+
+	const imgSrcCache = [];
+	const boxElmCache = [];
+	Array.from(document.querySelectorAll("[data-boxId]")).forEach((elm) => {
```
Comment (sculove) [inline demo/assets/js/carousel.js]: How about switching to the `map` method?
Semantically, map is better than forEach.

File: demo/demo.md [hunk 6]
```diff
@@ -87,3 +87,11 @@ var axes = new eg.Axes({
 	"rotateY": 315
 }, 100);
 ```
+
+### Carousel
+
+{% include_relative assets/html/carousel.html %}
+
+```js
+// Insert code.
```
Comment (sculove) [inline demo/demo.md]: Please put the main code for using axes

File: demo/assets/js/carousel.js [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+                boxElmCache[idx + 1].style[eg.Axes.TRANSFORM] = `scale(1)`;
+            }
+
+            if (idx === LAST_PANEL_IDX + 1) {
+                boxElmCache[idx - 1].classList.add("box-inactive");
+                boxElmCache[idx - 1].style[eg.Axes.TRANSFORM] = `scale(1)`;
+            }
+        },
+
+		"release": ({depaPos, destPos}) => {
```
Comment (sculove) [inline demo/assets/js/carousel.js]: remove `depaPos`

File: demo/assets/js/carousel.js (hunk 1 above)
Comment (JaeYeopHan) [inline demo/assets/js/carousel.js]: I forgot to erase it. Thx

File: demo/demo.md (hunk 6 above)
Comment (JaeYeopHan) [inline demo/demo.md]: I will modify it based on the code review and add it.

**Comment by JaeYeopHan** [comment]
Would I like to add a commit named `fix for review - [number]` and squash commits later so that you can see only what has changed?

File: demo/assets/js/carousel.js (hunk 7 above)
Comment (JaeYeopHan) [inline demo/assets/js/carousel.js]: I'll fix it

File: demo/assets/js/carousel.js (hunk 5 above)
Comment (JaeYeopHan) [inline demo/assets/js/carousel.js]: Good point :)

File: demo/assets/js/carousel.js (hunk 4 above)
Comment (JaeYeopHan) [inline demo/assets/js/carousel.js]: Good point :)

File: demo/assets/js/carousel.js (hunk 3 above)
Comment (JaeYeopHan) [inline demo/assets/js/carousel.js]: Oh, I did not know, thank you.

File: demo/assets/js/carousel.js (hunk 2 above)
Comment (JaeYeopHan) [inline demo/assets/js/carousel.js]: I'll try it

File: demo/assets/js/carousel.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    const removeSpotlight = (...indexes) => {
+        indexes.filter(idx => boxElmCache[idx]).forEach(idx => {
+            boxElmCache[idx].classList.add("box-inactive");
+            boxElmCache[idx].style[eg.Axes.TRANSFORM] = `scale(1)`;
+        });
+    }
+
+    axes.on({
+        "hold": ({inputEvent}) => {
+            if (inputEvent.target.parentNode.parentNode.id === "carousel-area") {
```
Comment (JaeYeopHan) [inline demo/assets/js/carousel.js]: Is there a better way to fire event in the "carousel-area" only?

**Comment by sculove** [comment]
@JaeYeopHan I will add  'demo' type. thanks
when I squashing I will take it!

File: demo/assets/css/carousel.css [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+  display: inline-block;
+  margin-top: 20px;
+  width: 48px;
+  padding: 8px;
+  border: 2px solid #F5C720;
+  border-radius: 50%;
+  font-size: 20px;
+  color: #F5C720;
+  text-align: center;
+}
```
Comment (sculove) [inline demo/assets/css/carousel.css]: put `user-select` in order to prevent selection

File: demo/demo.md (hunk 6 above)
Comment (sculove) [inline demo/demo.md]: How about animating when scaled ?
And I want the maximum scale to be bigger.

Comment (JaeYeopHan) [inline demo/demo.md]: I'll increase it to `1.7`.

File: demo/assets/js/carousel.js (hunk 8 above)
Comment (sculove) [inline demo/assets/js/carousel.js]: Hum...

**Comment by sculove** [comment]
LGTM

**Pull request merged**